projects/plugins/jetpack/extensions/blocks/writing-prompts/writing-prompts.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Anchor.fm integration.
+ *
+ * @since 9.3.0
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Extensions\WritingPrompts;
+
+use Automattic\Jetpack\Blocks;
+
+const FEATURE_NAME = 'writing-prompts';
+const BLOCK_NAME   = 'jetpack/' . FEATURE_NAME;
+
+/**
+ * Registers Anchor.fm integration for the block editor.
+ */
+function register_extension() {
+	Blocks::jetpack_register_block( BLOCK_NAME );
+}
+
+/**
+ * Checks URL params to determine the Anchor integration action to perform.
+ */
+function inject_writing_prompts() {
+	if (
+		! function_exists( 'get_current_screen' )
+		|| \get_current_screen() === null
+	) {
+		return;
+	}
+
+	// Return early if we are not in the block editor.
+	if ( ! wp_should_load_block_editor_scripts_and_styles() ) {
+		return;
+	}
+
+	$post = get_post();
+	if ( ! $post || ! $post->ID ) {
+		return;
+	}
+
```

Comment:
We need to only show this for `new` posts as well, currently it's showing when you edit an existing post.
<img width="344" alt="Screen Shot 2022-10-13 at 3 10 47 pm" src="https://user-images.githubusercontent.com/22446385/195507131-be785baf-e612-4f34-8ca8-927eb25a753c.png">